lib/ansible/plugins/action/synchronize.py

```diff
@@ -293,7 +294,10 @@ def run(self, tmp=None, task_vars=None):
             private_key = self._play_context.private_key_file
 
             if private_key is not None:
-                private_key = os.path.expanduser(private_key)
+                if dest_is_local:
+                    private_key = os.path.expanduser(private_key)
+                else:
+                    private_key = unfrackpath(private_key)
```

Comment:
Duh, sorry for wasting your time on that kind of thing, shouldn't have been rushing.

I've corrected that.